src/flamenco/runtime/program/fd_stake_program.c

```diff
@@ -2083,6 +2083,321 @@ redelegate( fd_exec_instr_ctx_t const * ctx,
   return 0;
 }
 
+// https://github.com/anza-xyz/agave/blob/cdff19c7807b006dd63429114fb1d9573bf74172/programs/stake/src/stake_state.rs#L136
+static int 
+move_stake_or_lamports_shared_checks( fd_exec_instr_ctx_t const *   invoke_context,
+                                      fd_borrowed_account_t  *      source_account,
+                                      ulong                         lamports,
+                                      fd_borrowed_account_t  *      destination_account,
+                                      ulong                         stake_authority_index,
+                                      fd_pubkey_t  const *          signers[static FD_TXN_SIG_MAX],
+                                      merge_kind_t *                source_merge_kind,
+                                      merge_kind_t *                destination_merge_kind,
+                                      uint *                        custom_err ) {
+    int rc;
+
+    // https://github.com/anza-xyz/agave/blob/cdff19c7807b006dd63429114fb1d9573bf74172/programs/stake/src/stake_state.rs#L145-L153
+    fd_pubkey_t const * stake_authority_pubkey = &invoke_context->instr->acct_pubkeys[stake_authority_index];
+    if( FD_UNLIKELY( !fd_instr_signers_contains( signers, stake_authority_pubkey ) ) ) {
```

Comment:
use is_signer_idx. This iterates through all of the accounts in the instruction and does memcmps